src/models/User.ts

```diff
@@ -179,9 +180,14 @@ export const userRouter = router({
 
       return result;
     }),
-  getProfile: publicProcedure.query(async ({ ctx }) =>
-    getProfile(ctx.state.storage)
-  ),
+  getProfile: publicProcedure.query(async ({ ctx }) => {
+    const profile = await getProfile(ctx.state.storage);
+    if (!profile) {
+      throw new Error();
```

Comment:
should this be the `NoUserFoundError`?

Reply:
Yes, think it should.. I've been gradually changing so that the profile will be populated at the first successful auth so it would make sense